doc/en/user/source/extensions/netcdf/netcdf.rst

```diff
@@ -345,4 +345,25 @@ parameters will have to be provided in a second property file, along with the li
 
 The tool supports other options as well, they can be discovered by running the tool without any parameter::
 
-  java -cp <path-to-geoserver>/WEB-INF/lib/*.jar org.geotools.coverage.io.netcdf.tools.H2Migrate -m <path-to-mosaic-directory> -is <indexPropertyFile> -isn <storeNameForIndex> -v
\ No newline at end of file
+  java -cp <path-to-geoserver>/WEB-INF/lib/*.jar org.geotools.coverage.io.netcdf.tools.H2Migrate -m <path-to-mosaic-directory> -is <indexPropertyFile> -isn <storeNameForIndex> -v
+  
+Caching
+-------
+When opening a NetCDF file, metadata and structures need to be setup, such as the Coordinate Reference System and related Coordinate Systems, the optional datastore configuration, the coverages structure (schemas and dimensions). Depending on the complexity of the file itself, those can be time consuming tasks. Operations that are continuously and repeatedly access the same files will be impacted by that. Therefore, starting with GeoServer 2.20.x a caching mechanism has been setup. 
+
+Some entities that can be considered *static* are internally cached once parsed: they include the netcdf datastore configuration setup on top of the datastore properties file, the indexer built on top of the auxiliary xml file, as well as the unit of measure of the variables. 
+
+.. note:: Make sure to do a GeoServer reload if one of these config files get modified or updated, to make sure that the cache get cleaned and the configs get updated.
+
+File Caching
+^^^^^^^^^^^^
+An additional level of caching can be manually enabled, so that NetCDF Files can be cached and re-used. Actually, the object being cached is not the whole file but a NetcdfDataset object, which is built on top of parsed metadata including coordinate system info. Whenever a NetCDF dataset is being accessed, a cached instance is provided and released back to the cache-pool once done. So if there are 10 concurrent requests accessing the same NetCDF file, up to 10 different NetCDF dataset cached instances will be used.
```

Comment:
```suggestion
An additional level of caching can be manually enabled, so that NetCDF Files can be cached and re-used. The object being cached is not the whole file, but a ``NetcdfDataset`` object, which is built on top of parsed metadata, including coordinate system info. Whenever a NetCDF dataset is being accessed, a cached instance is provided and released back to the cache-pool once done. So if there are 10 concurrent requests accessing the same NetCDF file, up to 10 different NetCDF dataset cached instances will be used.
```